lib/src/decorators/widget_decorator_widget.dart

```diff
@@ -62,6 +62,9 @@ class RenderWidgetDecorators extends StatelessWidget {
       final decorator = decoratorMap.remove(decoratorType);
       if (decorator == null) continue;
       current = decorator.build(mix, current);
+
+      // Remove the decorator from the MixData, because it can be applied only once.
+      mix.removeType(decoratorType);
```

Comment:
@tilucasoli I believe we might have to take another approach than mutating the attributes of the MixData; I think we might have to create an immutable class and pass it down to the child.

Reply:
Thanks, it's really important to remember! 
Thinking about possible solutions, I identified two. The first one is to propagate a new value in `RenderWidgetDecorators` using `MixProvider`. The second possibility is to create a new `MixData` without the decorator attributes when the existing `MixData` is inherited from `MixProvider`. My favorite solution is the second one, because adding a new widget to the widget tree unnecessarily doesn't seem like a good ideia.